src/app/components/ATIAnalytics/atiUrl/index.ts

```diff
@@ -332,3 +332,112 @@ export const buildATIEventTrackUrl = ({
     eventTrackingBeaconValues,
   )}&type=AT`;
 };
+
+export const buildReverbAnalyticsModel = ({
+  appName,
+  campaigns,
+  categoryName,
+  contentId,
+  contentType,
+  language,
+  ldpThingIds,
+  ldpThingLabels,
+  libraryVersion,
+  pageIdentifier,
+  pageTitle,
+  platform,
+  previousPath,
+  producerName,
+  origin,
+  nationsProducer,
+  statsDestination,
+  timePublished,
+  timeUpdated,
+}: ATIPageTrackingProps) => {
+  const href = getHref(platform);
+  const referrer = getReferrer(platform, origin, previousPath);
+
+  const aggregatedCampaigns = (Array.isArray(campaigns) ? campaigns : [])
+    .map(({ campaignName }) => campaignName)
+    .join('~');
+
+  const hashedId = getAtUserId();
+
+  const reverbVariables = {
+    params: {
+      page: {
+        contentId,
+        contentType,
+        destination: statsDestination,
+        name: pageIdentifier,
+        producer: producerName,
+        additionalProperties: {
+          app_name: platform === 'app' ? `${appName}-app` : appName,
+          app_type: getAppType(platform),
+          content_language: language,
+          product_platform: onOnionTld() ? 'tor-bbc' : null,
+          referrer_url:
+            referrer && encodeURIComponent(encodeURIComponent(referrer)),
+          x5: href && encodeURIComponent(encodeURIComponent(href)),
+          x8: libraryVersion,
+          x9: sanitise(pageTitle),
+          x10: nationsProducer && nationsProducer,
+          x11: timePublished,
+          x12: timeUpdated,
+          x13: ldpThingLabels,
+          x14: ldpThingIds,
+          x16: aggregatedCampaigns,
+          x17: categoryName,
+          x18: isLocServeCookieSet(),
+        },
+      },
+      user: {
+        hashedId,
+        isSignedIn: !!hashedId,
```

Comment:
@alex-magana I updated this functionality (see https://github.com/bbc/simorgh/pull/11353/commits/5321955a7f4f5b862c189877af5ec41d72bf1e30) because it allows us to tie user & visitor information together in Piano, which we need for future personalisation

Reply:
Since `hashedId` is always set to a value, in which case `!!hashedId` evaluates to `true`
for `isSignedIn`, what's the effect of setting `isSignedIn` to true prior to the roll-out of
personalisation?
- Is there any risk of data corruption associated with passing `isSignedIn` as `true` in Reverb?

Reply:
Ah I didn't realise `hashedId` would always have a value - I thought it might be `null` if a user declined tracking cookies? We might need to think about this more carefully when we productionise this. 

Also, good question RE setting `isSignedIn` to true. When I was digging about in the WebCore implementation, they set this value to true when a user is signed in to their BBC Account & Reverb sent the correct user ID values in the API request. Perhaps we need to restrict this to certain services - but I will eventually need to set this value to true my for ML project, in order to allow us to be able to query visit history in Piano.